nixos/modules/services/misc/mautrix-whatsapp.nix

```diff
@@ -0,0 +1,96 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mautrix-whatsapp;
+
+  configFile = pkgs.runCommand "mautrix-whatsapp" {
+    buildInputs = [ pkgs.mautrix-whatsapp pkgs.remarshal ];
+    preferLocalBuild = true;
+  } ''
+    mkdir -p $out
+    remarshal -if json -of yaml \
+      < ${pkgs.writeText "config.json" (builtins.toJSON cfg.configOptions)} \
+      > $out/config.yaml
+
+    ${pkgs.mautrix-whatsapp}/bin/mautrix-whatsapp -c $out/config.yaml -g -r $out/registration.yaml
+  '';
+
+in
+{
+  options.services.mautrix-whatsapp = {
+    enable = mkEnableOption "Mautrix-whatsapp, a puppeting bridge between Matrix and WhatsApp.";
+
+    configOptions = mkOption {
+      type = types.attrs;
+      description = ''
+        This options will be transform in YAML configuration file for the bridge
+
+        Look <link xlink:href="https://github.com/tulir/mautrix-whatsapp/wiki/Bridge-setup">here</link> for documentation.
+      '';
+      example = ''
```

Comment:
@Ma27 Should I put my configuration in example or in default ? If I use a default configuration this could be easier to use a "meta" module for all matrix's bridges.

Reply:
I'd say we keep place this into an example because defaults usually indicate that it's a sensitive value that works out of the box IMHO, but you have to tweak the config to get it working.
Also you use an attribute set as example, then you'll have syntax highlighting for free in the manual :)

Reply:
In Whatsapp case, we haven't a lot of value to customize : you can use the url of matrix-synapse module, use a port that we can test before to be sure it's not use and tadaaa :-) works out of the box

Reply:
True, but there are two reasons why I still think that it's better to keep this as example:

* Some mandatory values like HS tokens need to be set manually. For instance when I see a `default` value that appears reasonable to me I don't try to tweak it at first and then people might be confused when it breaks at runtime.
* It's harder to unset options than to declare new ones. For instance when using `postgres` instead of `sqlite` for persistence, you'd have to unset `uri` using `mkForce`.

Reply:
Ho yes you right.